### ISSUE TITLE
Char type

### ISSUE THREAD

**Issue by MichaelRFairhurst** [MichaelRFairhurst/wake-compiler#70]
Need to support a 'Char' type. I think I want to do it differently, like

    var a = a';
    var b = \n;

ie lexer rule ^(\\\w|\w')$

**Comment by MichaelRFairhurst** [comment]
While I like the brevity of this syntax, there are a few problems.

How do you write a space? ` '`? That is ambiguous in `'hello' + '....`. And what about a plus? `'hello'+'...`.

We could go with a'c to make it more specific, but it still might have issues where a or c are symbols, ie, `'hello'+'captain'`.And then it is still 3 characters, making it no better than the c style `'n'` (barring the fact that you can use single quote strings).

I do like \n still, especially for \{0-255}...

I wonder if I could make all single char strings into chars, and add a new literal for when they are indeed strings:

    'a' // char
    "a" // char
    '' // string
    "" // stirng
    "ab" // string
    'ab' // string
    'a's // string
    "a" // string

seems like the best solution

**Comment by MichaelRFairhurst** [comment]
oh man one realllly nice part of the \ syntax is the option to do

    \'

and

    \"

think of it!

    Printer.printLine("You entered");
    Printer.print(\" + name + \" + \n);

I do really like that syntax

**Comment by MichaelRFairhurst** [comment]
This is nearly done

it is worth noting that this comes up more often than expected.

Its usually OK, as in, `File.write(myvar + ":" + othervar)` where concatenation works, or even in `File.write(":")` where a Char is accepted, but it does require more code to handle Char inputs, and it can still fail with 'var Text filename = '_'; // prefix it` where you really are building a string but it starts with one val. It also comes up with `if Text == ' ' {`.

Granted `':'s` is always just a Char away (pun intended), but it might be best if char literals were treated specially; usable as a string or a char. Only once assigned in a particular manner would they truly be a Char or a String. However that seems like overthinking it....

**Comment by MichaelRFairhurst** [comment]
I will close this and make a new issue
